### DISCUSSION TITLE
Version 0.25.0

### DISCUSSION THREAD

**Discussion by jonasvoelcker** [tchiotludo/akhq#1644] [General]
Hey @tchiotludo,

we did some checks against our DEV-Cluster and found out that the current dev-Branch performs way better than version 0.24.0, might it be possible to release the current version soon?

Best Regards
Jonas

**Comment by AlexisSouquiere** [comment]
I agree too. And it's a huge one 😄 
I think PR #1650, #1654 and #1655 should be merged first but on my side I don't see any blocking bugs remaining that is preventing a release.

**Reply by alwibrm** [reply]
Hi @AlexisSouquiere, yes #1655 should definitely be in the next version. I'm looking forward to try 0.25.0, too.

**Reply by jaceko** [reply]
Agreed! I see that the dev branch has a lot of the security vulnerabilities resolved. Please release it.

**Reply by arindampatra33** [reply]
@tchiotludo yes please release a new version :)

**Reply by tchiotludo** [reply]
it's really near, sorry for the very late delay on that, but you know hard to find time!

**Reply by jonasvoelcker** [reply]
Hey @tchiotludo,

thanks a lot for taking that bunch of time and even if we push you here we are really greatful for your work :)

**Reply by raphael-afanyan** [reply]
Hi @tchiotludo, thanks for all the work you are doing, can we expect a new version soon?

**Reply by yanntoque** [reply]
Hello, 
Any news about a release date for the 0.25.0, i can't image the work behind it. 
I'm not here to complain, but as we use your solution in 0.24.0 (released on April 2023) and there are a bunch of CVEs. 
We are sadly thinking about using another solution to mitigate those CVEs to be compliant.

**Reply by AlexisSouquiere** [reply]
@yanntoque the PR #1678 is in its final stage. Development is completed. I deployed it in my company on a panel of users to get feedback. If there is no blockers in the coming days I will mark it as ready and then release can be done quickly. I would still recommend to let people try the dev branch for a week or two and see if we didn't miss anything.
Anyway, feel free to contribute to fix the CVEs in the future ;)

**Reply by jonasvoelcker** [reply]
Hey @yanntoque,

we are still busy tackling the frontend rework to resolve many of those CVEs. @AlexisSouquiere just started to test the version we prepared on their dev cluster. I did also manage to check the basic functions against our Dev Kafka.

Do you mind checking out our branch and providing your feedback? :)

https://github.com/tchiotludo/akhq/pull/1678

Best Regards
Jonas

**Comment by jonasvoelcker** [comment]
Hi @tchiotludo, do you maybe have some checklist what needs to be done before releasing? Maybe some of us could add some value but I guess nobody knows, what's in your mind ;)

**Reply by AlexisSouquiere** [reply]
We discussed in #1472 about JS dependencies upgrade. I started it in #1678 but it's indeed a long task. I think the task is 60% finished but if some of you are skilled in React or can do non-regression testing, help is welcome 😄 
Almost all the screens are working fine, some CSS is still broken on some screen but fixing the remaining issue and testing will take some time. UI got a small facelift:

<img width="1690" alt="image" src="https://github.com/tchiotludo/akhq/assets/2262145/cc93f71e-0085-428c-b40e-ba0d0783b954">

**Reply by jonasvoelcker** [reply]
Hey @AlexisSouquiere, I just took some time to run the branch locally and my honest opinion is that it is pretty far from being a release. The old implementation uses many custom styles and combined with the new bootstrap-library it results in many paddings and colors being wrong. We may find a way how I can support on this but it should not hinder the new version from being released as it will definitely take many weeks from now to get the UI out. @tchiotludo Are there additional reasons for the JS-Libraries being updated? I suppose that the current versions will be no downgrade compared with 0.24.0 ;)

**Reply by AlexisSouquiere** [reply]
I understand your point and also @tchiotludo wish to upgrade the frontend dependencies. It hasn't been done for a while and delaying it once again will increase the workload in the future to fill the gap.
When I said that it was 60% completed it was mainly because there were breaking changes especially in react-router and old custom styles. I started to deal with styling recently. There are still styles to fix here and there but I found the UI lighter with the standard bootstrap style and only few adjustments (it's personal). 
I have no issues releasing a new version with or without the new UI. In any case if someone wants to help, he/she is welcome !

**Reply by tchiotludo** [reply]
@jonasvoelcker, we have someone really motivated to work on the UI upgrade, it's something really valuable, and thanks @AlexisSouquiere to take the subject. 
I already looked month ago, and the work amount is really huge but it needed to be done. AKHQ is used in large corporate with CVE scanner, and there is a lots because of the frontend, If we can remove this one, all the community will be happy.  
I know that delayed a release can be painful but I think it's not a pain to wait some weeks more. 
At the end, this release will be the biggest one and thanks to @AlexisSouquiere for large parts.

**Reply by AlexisSouquiere** [reply]
And also to @jonasvoelcker who is the one helping me (and did a loooot) on the UI upgrade 😉

**Reply by shaneikennedy** [reply]
@tchiotludo or @AlexisSouquiere Any chance there's a checklist of what's left before release? Happy to help out 🙂

**Reply by jonasvoelcker** [reply]
Hey @shaneikennedy,

we just managed to merge all the stuff together today. So there will not be much left, I think.

What I do not understand is the failing action in the Main-Workflow:

cp: target 'build/distributions/akhq.tar' is not a directory

If you would find a solution for that, it would be really great 😉

**Reply by shaneikennedy** [reply]
https://github.com/tchiotludo/akhq/issues/1702

it was me who broke it ! definitely revert my commit if that's all that's blocking a release 👍

**Comment by jonasvoelcker** [comment]
Unfortunately there is still at least this issue blocking the current version being released: https://github.com/tchiotludo/akhq/issues/1704

It causes every local setup without enabled security to break.

**Reply by shaneikennedy** [reply]
How about now? Happy to help out if I can :)

**Reply by jonasvoelcker** [reply]
This one should also be merged: https://github.com/tchiotludo/akhq/pull/1708

The dependabot changes can also be merged if @tchiotludo runs this command first:

@dependabot rebase

Afterwards we are ready to go in my opinion.

**Comment by tchiotludo** [comment]
@AlexisSouquiere @jonasvoelcker I've updated to latest dependencies yesterday. 
Only the confluentVersion couldn't be updated (missing dependencies in kSQL, don't know why). 
If you have time, you make a final review on the `dev` image, I will published the new version just after

**Reply by jonasvoelcker** [reply]
@alwibrm Could you please install the latest one so we can test it?

**Reply by jonasvoelcker** [reply]
Hey @tchiotludo, we found an issue with the after-parameter which is addressed by this PR: https://github.com/tchiotludo/akhq/pull/1754

Would you be so kind to merge it pretty soon so we can continue with testing?

**Reply by jonasvoelcker** [reply]
Hi @tchiotludo, are there any news?

**Reply by tchiotludo** [reply]
oups didn't see the PR pass, it's merged

**Reply by jonasvoelcker** [reply]
@tchiotludo There was still an implementation error addressed here: https://github.com/tchiotludo/akhq/pull/1757

**Reply by jonasvoelcker** [reply]
Now it seems to be working, great!

@alwibrm Do you agree that the version is ready to be released for us now?

**Reply by alwibrm** [reply]
@jonasvoelcker @tchiotludo Looks good to me, the search feature is working fine.

**Reply by jonasvoelcker** [reply]
@AlexisSouquiere @shaneikennedy @ebrard Did you also check on your side?

**Reply by ebrard** [reply]
we've been using the dev image for a while now (with the audit feature on), looks fine for me.

**Reply by AlexisSouquiere** [reply]
I was busy the last two weeks I didn't had time to install it. But I just deployed it and it seems fine for me.
Let's go for the release 🚀

**Reply by jonasvoelcker** [reply]
@tchiotludo Would you be so kind? 😃

**Reply by deibl** [reply]
Guess i might have to look for a new morning routine soon in the Future.
Fetch Coffee, Boot my Mac, Check if AKHQ 0.25.0 was released, proceed with my actual work.

**Reply by alwibrm** [reply]
@tchiotludo What do you think, are we ready to release 0.25.0?

**Reply by shaneikennedy** [reply]
@jonasvoelcker sorry for the slow reply i don't have my github notifcations setup well. I'd love to test this on our staging and testing clusters but we ship the akhq jar file that gets published on release, so I'd need an "official" release to do any non-trivial testing (without doing a ton of work, that is).

Happy to just release 0.25.0 and provide feedback and contributions as we test it, releasing doesn't mean everyone is forced to upgrade 👍

**Reply by jonasvoelcker** [reply]
@tchiotludo I agree with @shaneikennedy, nobody is forced to upgrade and even if we find new bugs that need to be fixed, there can easily be a version 0.25.1 soon. Is there anything which hinders you from releasing?

**Comment by theBNT** [comment]
Hi, the new release would also help the much anticipated JSON logging as described here PR #1772 <3

**Comment by alfsch** [comment]
It would be also fine to have a tagged pre-release which we could reference without using the dev tag.

**Comment by jonasvoelcker** [comment]
Eureka 😁
